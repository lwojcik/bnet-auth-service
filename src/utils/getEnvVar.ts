interface GetEnvVarArgument {
  appPrefix?: string;
  featurePrefix?: string;
  property: string;
}

export const getEnvVar = ({
  appPrefix,
  featurePrefix,
  property,
}: GetEnvVarArgument) =>
  `${appPrefix ? `${appPrefix}_` : ''}${
    featurePrefix ? `${featurePrefix}_` : ''
  }${property.toUpperCase()}`;
