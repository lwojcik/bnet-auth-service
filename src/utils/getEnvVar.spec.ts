import { getEnvVar } from './getEnvVar';

describe('getEnvVar', () => {
  const testAppPrefixes = ['TEST_APP1', 'TEST_APP2'];

  const testFeaturePrefixes = ['FEATURE1', 'FEATURE2', 'FEATURE3'];

  const testProperties = ['PROPERTY1', 'PROPERTY2', 'PROPERTY3'];

  it('should generate correct environment variable names', () => {
    testAppPrefixes.forEach((appPrefix) => {
      testFeaturePrefixes.forEach((featurePrefix) => {
        testProperties.forEach((property) => {
          expect(
            getEnvVar({ appPrefix, featurePrefix, property })
          ).toMatchSnapshot();

          expect(getEnvVar({ featurePrefix, property })).toMatchSnapshot();

          expect(getEnvVar({ property })).toMatchSnapshot();
        });
      });
    });
  });
});
