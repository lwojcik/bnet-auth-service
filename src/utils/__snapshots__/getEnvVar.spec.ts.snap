// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEnvVar should generate correct environment variable names 1`] = `"TEST_APP1_FEATURE1_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 2`] = `"FEATURE1_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 3`] = `"PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 4`] = `"TEST_APP1_FEATURE1_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 5`] = `"FEATURE1_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 6`] = `"PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 7`] = `"TEST_APP1_FEATURE1_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 8`] = `"FEATURE1_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 9`] = `"PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 10`] = `"TEST_APP1_FEATURE2_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 11`] = `"FEATURE2_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 12`] = `"PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 13`] = `"TEST_APP1_FEATURE2_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 14`] = `"FEATURE2_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 15`] = `"PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 16`] = `"TEST_APP1_FEATURE2_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 17`] = `"FEATURE2_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 18`] = `"PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 19`] = `"TEST_APP1_FEATURE3_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 20`] = `"FEATURE3_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 21`] = `"PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 22`] = `"TEST_APP1_FEATURE3_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 23`] = `"FEATURE3_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 24`] = `"PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 25`] = `"TEST_APP1_FEATURE3_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 26`] = `"FEATURE3_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 27`] = `"PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 28`] = `"TEST_APP2_FEATURE1_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 29`] = `"FEATURE1_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 30`] = `"PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 31`] = `"TEST_APP2_FEATURE1_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 32`] = `"FEATURE1_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 33`] = `"PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 34`] = `"TEST_APP2_FEATURE1_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 35`] = `"FEATURE1_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 36`] = `"PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 37`] = `"TEST_APP2_FEATURE2_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 38`] = `"FEATURE2_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 39`] = `"PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 40`] = `"TEST_APP2_FEATURE2_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 41`] = `"FEATURE2_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 42`] = `"PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 43`] = `"TEST_APP2_FEATURE2_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 44`] = `"FEATURE2_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 45`] = `"PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 46`] = `"TEST_APP2_FEATURE3_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 47`] = `"FEATURE3_PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 48`] = `"PROPERTY1"`;

exports[`getEnvVar should generate correct environment variable names 49`] = `"TEST_APP2_FEATURE3_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 50`] = `"FEATURE3_PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 51`] = `"PROPERTY2"`;

exports[`getEnvVar should generate correct environment variable names 52`] = `"TEST_APP2_FEATURE3_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 53`] = `"FEATURE3_PROPERTY3"`;

exports[`getEnvVar should generate correct environment variable names 54`] = `"PROPERTY3"`;
